### PR TITLE
Fix integration manager `get_open_id_token` action and add E2E tests

### PR DESCRIPTION
**Pull request by justinbot**
The `get_open_id_token` integration manager action added in #8782 was broken in a couple of ways:
* Missing an `await` so it always failed with this error:
    ```
    Unable to fetch openId token. DOMException: Failed to execute 'postMessage' on 'Window': #<Promise> could not be cloned.
    ```
* Mistakenly required user ID and room ID parameters 

This PR fixes those issues and adds an E2E test.

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix integration manager `get_open_id_token` action and add E2E tests ([\#9520](https://github.com/matrix-org/matrix-react-sdk/pull/9520)).

### CODE REVIEW

File: src/ScalarMessaging.ts [hunk 1]
```diff
@@ -715,6 +715,9 @@ const onMessage = function(event: MessageEvent<any>): void {
         } else if (event.data.action === Action.SetWidget) {
             setWidget(event, null);
             return;
+        } else if (event.data.action === Action.GetOpenIdToken) {
+            getOpenIdToken(event);
+            return;
```
Comment (richvdh) [inline src/ScalarMessaging.ts:720]: the comment on line 709 "Get and set user widgets" seems to be incomplete now. I don't think it adds any value.

File: cypress/e2e/integration-manager/get-openid-token.spec.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            }).as("roomId");
+        });
+    });
+
+    afterEach(() => {
+        cy.stopSynapse(synapse);
+        cy.stopWebServers();
+    });
+
+    it("should obtain an openID token", () => {
```
Comment (richvdh) [inline cypress/e2e/integration-manager/get-openid-token.spec.ts:129]:
```suggestion
    it("should successfully obtain an openID token", () => {
```

**Review by richvdh — COMMENTED** [review]
looks good to me otherwise!

**Comment by richvdh** [comment]
... except that the code coverage gate isn't passing, which seems surprising. Is the new test not running, or not being counted, for some reason?

**Comment by t3chguy** [comment]
@richvdh the test coverage is for jest tests, cypress tests don't count as even loading the app gets you like 25% coverage

**Review by richvdh — COMMENTED** [review]
the changes look good, but CI is failing...

**Review by turt2live — APPROVED** [review]
force-merging due to sonar not understanding cypress.

**Pull request merged**